### PR TITLE
end pass phrase session from the notification bar

### LINKED ISSUE #2228: Question: How to manually end an active passphrase session
The FlowCrypt browser extension has the option to manually end the passphrase session. 

As far as I know, the FlowCrypt android app doesn't have this option. So when the FlowCrypt android app users choose to use the remember passphrase temporarily option, the user would have to wait for the passphrase session to lapse or reboot the phone to force the app to ask for the passphrase after closing the app.

By having this option, we can give the users additional confidence when using the app. 

Reference: https://mail.google.com/mail/u/human@flowcrypt.com/#inbox/FMfcgzGrcrtWVLBsDHHQVzsmTHmKPdNt

### PR DESCRIPTION
**Pull request by DenBond7**
This PR Added ability to end pass phrase session from the notification bar

close #2228

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/fragment/isolation/incontainer/EndPassPhraseSessionInIsolationTest.kt [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+          .clazz("android.widget.Button")
+          .res("android:id/expand_button")
+      )
+
+      if (expandButton != null) {
+        expandButton.click()
+        it.findObject(By.text(endPassPhraseSessionLabel))
+      } else null
+    } else null
+  }
```
Comment (ioanmo226) [inline FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/fragment/isolation/incontainer/EndPassPhraseSessionInIsolationTest.kt:173]: IMO, better to like as belows.
For now, not sure about function return type and doesn't seem like best idea to use `= device.findObjects` in function implmentation.
Removed unnecessary `if (expandButton != null) {` check too though
What do you think?

```kt
private fun getActionButtonUiObject2(
  device: UiDevice,
  activePassPhraseSessionLabel: String,
  endPassPhraseSessionLabel: String
): UiObject2? {
  val notificationShadeObjects = device.findObjects(By.res(notificationResourcesName))

  return notificationShadeObjects.firstNotNullOfOrNull { notification ->
    if (notification.hasObject(By.text(activePassPhraseSessionLabel))) {
      val expandButton = notification.findObject(
        By.pkg("com.android.systemui")
          .clazz("android.widget.Button")
          .res("android:id/expand_button")
      )

      expandButton?.click()
      notification.findObject(By.text(endPassPhraseSessionLabel))
    } else null
  }
}
```

File: FlowCrypt/src/main/java/com/flowcrypt/email/security/KeysStorageImpl.kt [hunk 2]
```diff
@@ -224,8 +241,16 @@ class KeysStorageImpl private constructor(context: Context) : KeysStorage {
...
+  override fun hasEmptyPassphrase(vararg types: KeyEntity.PassphraseType): Boolean {
+    return passPhraseMap.values
+      .filter { it.passphraseType in types }
+      .any { it.passphrase.isEmpty }
+  }
+
+  override fun hasNonEmptyPassphrase(vararg types: KeyEntity.PassphraseType): Boolean {
+    return passPhraseMap.values
+      .filter { it.passphraseType in types }
+      .any { !it.passphrase.isEmpty }
```
Comment (ioanmo226) [inline FlowCrypt/src/main/java/com/flowcrypt/email/security/KeysStorageImpl.kt:253]: Same as above

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/fragment/isolation/incontainer/EndPassPhraseSessionInIsolationTest.kt (hunk 1 above)
Comment (DenBond7) [reply FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/fragment/isolation/incontainer/EndPassPhraseSessionInIsolationTest.kt]: I've added changes. Please relook.

**Review by ioanmo226 — APPROVED** [review]
👍

**Pull request merged**